flink-libraries/flink-table/src/test/scala/org/apache/flink/api/table/expressions/ScalarFunctionsTest.scala

```diff
@@ -1101,6 +1101,45 @@ class ScalarFunctionsTest extends ExpressionTestBase {
       "true")
   }
 
+  @Test
+  def testInExpressions(): Unit = {
+    testTableApi(
```

Comment:
@twalthr Are you sure? That we need use IN with POJOs/Tuples/Case classes?
First of all I will hit performance because compare of these types too complicated. it's easier get subset of id's and use IN than compare POJOs and Case classes. More over we cant it use in SQL string statements (we need parse POJO and case classes from string)